db/migrate/20210621002201_add_columns_to_email_log_to_match_incoming_for_smtp_imap.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class AddColumnsToEmailLogToMatchIncomingForSmtpImap < ActiveRecord::Migration[6.1]
+  def up
+    add_column :email_logs, :cc_addresses, :text, null: true
+    add_column :email_logs, :cc_user_ids, :integer, array: true, null: true
+    add_column :email_logs, :raw, :text, null: true
+    add_column :email_logs, :topic_id, :integer, null: true
+
+    add_index :email_logs, :topic_id
+  end
+
+  def down
+    remove_column :email_logs, :cc_addresses if column_exists?(:email_logs, :cc_addresses)
+    remove_column :email_logs, :cc_user_ids if column_exists?(:email_logs, :cc_user_ids)
+    remove_column :email_logs, :raw if column_exists?(:email_logs, :raw)
+    remove_column :email_logs, :topic_id if column_exists?(:email_logs, :topic_id)
+
+    remove_index :email_logs, :topic_id if index_exists?(:email_logs, [:topic_id])
```

Comment:
I think we can just raise an irreversible migration error here.

Reply:
It's not irreversible though? Can easily just remove the columns...

Reply:
The code here is only useful in development but once we decide on the data model, I feel like we don't need to carry this code long term since we will never rollback in production.

Reply:
Hmm I don't really agree with this; sure we don't rollback in production, but we don't worry about this when we do `change` migrations that have automatic rollbacks e.g. https://github.com/discourse/discourse/commit/1cd0424ccdb44eafed4e9df23c8b9013e9f127bf#diff-0e62f4a247b92c1618d42757e480f1477a7210cd12b620f05aaa0ceda9de395cR3

I think we need a tiebreaker :P

Reply:
I guess we keep it since the code has already been written :grin:

Reply:
As a tiebreaker I think it's fine to keep and should work.

I do think you could have done it as a `change` migration and it would automatically create the `down` code for you. Also your `down` is out of order. It should be the opposite of up. This means that your code to remove an index will never run because the column will already be deleted (and the index with it.) This is mostly aesthetic though - in the end your code works.